aws/resource_aws_dynamodb_table.go

```diff
@@ -652,7 +715,23 @@ func resourceAwsDynamoDbTableDelete(d *schema.ResourceData, meta interface{}) er
 
 	log.Printf("[DEBUG] DynamoDB delete table: %s", d.Id())
 
-	err := deleteAwsDynamoDbTable(d.Id(), conn)
+	input := &dynamodb.DescribeTableInput{
+		TableName: aws.String(d.Id()),
+	}
+
+	output, err := conn.DescribeTable(input)
+	log.Printf("[DEBUG] DynamoDB delete describe: %s", output)
+	if err != nil {
+		return fmt.Errorf("error describing DynamoDB Table (%s):  %s", d.Id(), err)
+	}
+
+	if len(output.Table.Replicas) > 0 {
+		if err := deleteDynamoDbReplicas(d.Id(), d.Get("replica").([]interface{}), conn); err != nil {
```

Comment:
We typically avoid additional re-reads in `Delete` functions and prefer to just skip any errors that indicate that any deletion already happened. If there are additional replicas added outside what was previously refreshed into the Terraform state, we _should_ allow the API to return an error so the operator can determine if the deletion is still safe.

```suggestion
	if replicas := d.Get("replica").(*schema.Set).List(); len(replicas) > 0 {
			if err := deleteDynamoDbReplicas(d.Id(), replicas, conn); err != nil {
```